stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1652,6 +1653,27 @@ public enum TensorSliceIndex : TensorSliceIndexProtocol {
   public var sliceIndex: TensorSliceIndex { return self }
 }
 
+extension TensorSliceIndex: Equatable {
+  public static func == (lhs: TensorSliceIndex, rhs: TensorSliceIndex) -> Bool {
```

Comment:
Did the compiler actually ask you to implement this yourself? The compiler should derive this automatically since all associated values are already equatable.

Reply:
They're not actually. For example, none of the partial ranges are equatable.

Reply:
I see!